pkg/hypervctl/vmm.go

```diff
@@ -88,31 +89,57 @@ func (vmm *VirtualMachineManager) Exists(name string) (bool, error) {
 	return false, nil
 }
 
+// GetMachine is a stub to lookup and get settings for a VM
 func (vmm *VirtualMachineManager) GetMachine(name string) (*VirtualMachine, error) {
+	_, vm, err := vmm.getMachine(name)
+	return vm, err
+}
+
+// GetMachineExists is the same as GetMachine but also returns a bool to distinguish between
+// lookup errors (permissions, etc) and the machine really does not exist.  If the machine truly does
+// not exist in hyperv, the result of the returned error will be nil.  i.e. a signature of false, nil, nil
+func (vmm *VirtualMachineManager) GetMachineExists(name string) (bool, *VirtualMachine, error) {
+	exists, vm, err := vmm.getMachine(name)
+	if errors.Is(err, wmiext.ErrNoResults) {
+		err = nil
+	}
+	return exists, vm, err
+}
+
+// getMachine looks up a single VM by name and returns a bool if the VM exists
+// AND can be queried for its settings.
+func (vmm *VirtualMachineManager) getMachine(name string) (bool, *VirtualMachine, error) {
```

Comment:
```suggestion
func (vmm *VirtualMachineManager) getMachine(name string) (*VirtualMachine, error) {
```

+ changes that remove the returned bool